client-java/src/main/java/uk/gov/gchq/palisade/client/util/Checks.java

```diff
@@ -73,7 +73,7 @@
      * @return the argument
      * @throws IllegalArgumentException if {@code argument} is null
      */
-    static <T> T checkArgument(final T argument, final String template, final Object... args) {
+    static <T> T checkNotNull(final T argument, final String template, final Object... args) {
         if (argument == null) {
             throw new IllegalArgumentException(format(template, args));
```

Comment:
That `format` method is a private method in the class. If a problem occurs because someone has mis-configured the number of parameters, the check should not fail. It will just use the template as is, but still throw the required exception.